js/src/index.ts

```diff
@@ -6,36 +6,51 @@
 //
 //
 
-import { 
-    Connection, 
+import {
+    Connection,
     Keypair,
-    PublicKey, 
+    PublicKey
 } from '@solana/web3.js';
-import { 
-    NautilusQuery, 
+import { NautilusIdl, NautilusProgram } from './types';
+import {
+    NautilusQuery,
     NautilusTable,
 } from './sql';
-import { NautilusUtils } from './util';
 
-export class Nautilus {
+import { decapitalizeFirstLetter } from './util';
+
+export class Nautilus<Program extends NautilusProgram = NautilusProgram> {
 
     connection: Connection;
-    programs: [PublicKey, string][];
-    defaultProgram: PublicKey | undefined;
-    payer: Keypair | undefined;
+    programId: PublicKey;
+    programs?: {[programName: string]: PublicKey};
+    payer?: Keypair | undefined;
 
-    util: NautilusUtils = new NautilusUtils();
+    readonly idl: NautilusIdl;
+    readonly tables: { [N in keyof Program["tables"]]: NautilusTable };
 
-    constructor(
+    constructor ({ connection, idl, programId, programs, payer }: {
         connection: Connection,
-        programs: string | PublicKey | [PublicKey, string][],
-        defaultProgram?: string | PublicKey,
+        idl: NautilusIdl,
+        programId: string | PublicKey,
+        programs?: {[programName: string]: PublicKey},
         payer?: Keypair,
-    ) {
-        
+    }) {
+
         this.connection = connection;
-        [this.programs, this.defaultProgram] = parseNautilusArgs(programs, defaultProgram)
+        this.programId = programId instanceof PublicKey ? programId : new PublicKey(programId);
+        this.programs = programs
         this.payer = payer ?? undefined;
+
+        this.idl = idl
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any

```

Comment:
I see why you need this annotation but I'm curious if we can use `Object` instead somehow?

Reply:
I will dig deeper into this, I used the annotation as a workaround but forgot to add a // TODO